modules/calib/src/calibration.cpp

```diff
@@ -1045,29 +1052,44 @@ static double stereoCalibrateImpl(
                     perViewErr->at<double>(i, k) = std::sqrt(viewErr/ni);
                 reprojErr += viewErr;
             }
+
+            ptPos += ni;
         }
-        if(_errnorm)
-            *_errnorm = reprojErr;
+        errnorm = reprojErr;
         return true;
-    });
+    };
+
+    double reprojErr = 0;
+    if (countNonZero(mask))
+    {
+        LevMarqDenseLinear solver(param, lmcallback, mask);
+        // old LMSolver calculates successful iterations only, this one calculates all iterations
+        solver.maxIterations = (unsigned int)(termCrit.maxCount * 2.1);
+        solver.stepNormTolerance = termCrit.epsilon;
+        solver.smallEnergyTolerance = termCrit.epsilon * termCrit.epsilon;
```

Comment:
We should have setter with TermCriteria parameter.

> maxIterations = (unsigned int)(termCrit.maxCount * 2.1);
> 2.1

Why?

Reply:
`2.1` is a compatibility hack. Old impl counts successful iterations only, the new one counts all iterations.
The proportion between them for the same run is different but in average it is 2.1: `newIters = oldIters*2.1`

I propose three solutions here:
* to leave this hack as is (adding a comment about it in src)
* to find true multiplier for each use case based on iterations elapsed statistics
* remove the multiplier: it's 5.x now, we don't have to maintain such parameters compatibility

Reply:
I think, having such a hack is fine.
1. If the problem solved is more or less well-defined (not ill-posed), the solver should converge faster than it will reach the maximum. 
2. If the problem solved is ill-posed then we will do at least termCrit.maxCount iterations, so this hack may affect the speed, but not the quality. But, I believe, because the new solver uses the sparse structure of matrices, it should run even faster than the previous "dense" implementation.

Reply:
Multiplier removed, statistics recalculated.
Regarding `TermCriteria` usage: I'm against it, it has only one epsilon parameter and it's not obvious to what threshold in this LevMarq impl it corresponds to.